internal/config/instance.go

```diff
@@ -177,11 +167,11 @@ func Get() (*Instance, error) {
 	return defaultConfig, nil
 }
 
-// SetWithLock updates a value at the given key. The valueF argument returns the
+// Update updates a value at the given key. The valueF argument returns the
 // new value based on the previous one.  If the function returns with an error, the
 // update is cancelled.  The function ensures that no-other process or thread can modify
 // the key between reading of the old value and setting the new value.
-func (i *Instance) SetWithLock(key string, valueF func(oldvalue interface{}) (interface{}, error)) error {
+func (i *Instance) Update(key string, valueF func(oldvalue interface{}) (interface{}, error)) error {
```

Comment:
I know I'm not flagged for review, but I want to ensure this isn't merged as is. 

I see two issues here.

  1. I expressed my issues with this function even existing. Why then rename it to make it even more prone to adoption?

  2. This struct already has a `Set()` method. Introducing a vague `Update()` analog is just confusing and inconsistent.

Primarily the first though. The fact that this was renamed makes me concerned that my concerns about this function were not heard or understood. To be clear: this function is (and SHOULD be) an eyesore. I want to get rid of it as soon as reasonably possible. That means minimizing its effect on our codebase and exploring better alternatives.